backend/tasks/tasks.py

```diff
@@ -45,7 +57,108 @@ class Lead:
     @property
     def formation_date_str(self):
         return self.formation_date.strftime(DATE_FORMAT)
+    
+def find_CNP_leads() -> int:
+    inserted = 0
+    entities = csv_to_entities('CNPData.csv')
+    
+    
+    leads = (
+        _map_CNP_entity_to_lead(entity)
+        for entity in entities
+    )
+    
+    with db.get_engine().begin() as connection:
+        for lead in leads:
+            connection.execute(text(
+                """
+                    INSERT INTO leads (
+                        company_name,
+                        company_address,
+                        formation_date,
+                        website,
+                        phone,
+                        email,
+                        facebook,
+                        data_source
+                    )
+                    SELECT
+                        :company_name,
+                        :company_address,
+                        :formation_date,
+                        :website,
+                        :phone,
+                        :email,
+                        :facebook,
+                        'ColoradoNonprofitAssociation'
+                    WHERE NOT EXISTS (
+                        SELECT 1 FROM leads
+                        WHERE company_name = :company_name
+                    )
+                """),
+                company_name=lead.company_name,
+                company_address=lead.company_address,
+                formation_date=lead.formation_date_str,
+                website=lead.website,
+                phone=lead.phone,
+                email=lead.email,
+                facebook=lead.facebook,
+            )
+            inserted += 1
+    return inserted
+
+def _map_CNP_entity_to_lead(entity: CNPBusinessEntity) -> Lead:
+    return Lead(
+        company_name=entity.entityname,
+        company_address=entity.principaladdress1,
+        website=entity.website,
+        phone=entity.phone,
+        email=entity.email,
+        facebook=entity.facebook,
+        formation_date=datetime.strptime(
+            entity.entityformdate.split('T')[0], DATE_FORMAT),
+    )
+
+def csv_to_entities(file_name):
+    
+    with open(file_name) as csv_file:
+        
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        line_count = 0
+        business_entities = []
+        for row in csv_reader:
+            entity = CNP_row_to_dict(row)
```

Comment:
Check out [csv.DictReader](https://docs.python.org/3/library/csv.html#csv.DictReader)

I think it will streamline a lot of the row to dict conversion code.